src/DotNet.Status.Web/Controllers/AzurePipelinesController.cs

```diff
@@ -0,0 +1,375 @@
+                    continue;
+
+                if (monitor.Branches.All(mb => !string.Equals($"{fullBranchPrefix}{mb}",
+                    build.SourceBranch,
+                    StringComparison.OrdinalIgnoreCase)))
+                {
+                    continue;
+                }
+
+                string prettyBranch = build.SourceBranch;
+                if (prettyBranch.StartsWith(fullBranchPrefix))
+                {
+                    prettyBranch = prettyBranch.Substring(fullBranchPrefix.Length);
+                }
+
+                _logger.LogInformation(
+                    "Build '{buildNumber}' in project '{projectName}' with definition '{definitionPath}' and branch '{branch}' matches monitoring criteria, sending notification",
+                    build.BuildNumber,
+                    build.Project.Name,
+                    build.Definition.Path,
+                    build.SourceBranch);
+                
+                _logger.LogInformation("Fetching timeline messages...");
+                string timelineMessage = await BuildTimelineMessage(build);
+                
+                _logger.LogInformation("Fetching changes messages...");
+                string changesMessage = await BuildChangesMessage(build);
+
+                BuildMonitorIssuesOptions repo = _options.Value.Issues;
+                IGitHubClient github = await _gitHubClientFactory.CreateGitHubClientAsync(repo.Owner, repo.Name);
+                
+                DateTimeOffset? finishTime = DateTimeOffset.TryParse(build.FinishTime, out var parsedFinishTime) ?parsedFinishTime: (DateTimeOffset?) null;
+                DateTimeOffset? startTime = DateTimeOffset.TryParse(build.StartTime, out var parsedStartTime) ? parsedStartTime:(DateTimeOffset?) null;
+
+                string timeString = "";
+                string durationString = "";
+                if (finishTime.HasValue)
+                {
+                    timeString = finishTime.Value.ToString("R");
+                    if (startTime.HasValue)
+                    {
+                        durationString = ((int) (finishTime.Value - startTime.Value).TotalMinutes) + " minutes";
+                    }
+                }
+
+                string icon = build.Result == "failed" ? ":x:" : ":warning:";
+
+                string body = @$"Build [#{build.BuildNumber}]({build.Links.Web.Href}) {build.Result}
+
+## {icon} : {build.Project.Name} / {build.Definition.Name} {build.Result}
+
+### Summary
+**Finished** - {timeString}
+**Duration** - {durationString}
+**Requested for** - {build.RequestedFor.DisplayName}
+**Reason** - {build.Reason}
+
+### Details
+
+{timelineMessage}
+
+### Changes
+
+{changesMessage}
+";
+
+                var newIssue =
+                    new NewIssue($"Build failed: {build.Definition.Name}/{prettyBranch} #{build.BuildNumber}")
```

Comment:
Kind of weird name for a class: `NewIssue`

Reply:
It's not my name. That class comes from "Octokit".  It's siblings are "UpdateIssue" (that you pass to Issue.Update) and just "Issue", which is the value returned from Issue.Get